src/Core/Geographic/Coordinates.js

```diff
@@ -1,441 +1,270 @@
+                ellipsoid.cartesianToCartographic(this, target);
+            } else {
+                this.as('EPSG:4326', coordPivot);
+                target.set(proj4cache('EPSG:4326', crs).forward([coordPivot.x, coordPivot.y]), coordPivot.z);
+            }
+        } else if (CRS.is4326(this.crs) && crs == 'EPSG:3857') {
+            this.y = THREE.Math.clamp(this.y, -89.999999, 89.999999);
+            target.set(proj4cache(this.crs, crs).forward([this.x, this.y]), this.z);
+        } else {
+            target.set(proj4cache(this.crs, crs).forward([this.x, this.y]), this.z);
+        }
 
-/**
- * Returns coordinates in the wanted [CRS]{@link http://inspire.ec.europa.eu/theme/rs}.
- * @example
- *
- * const position = { longitude: 2.33, latitude: 48.24, altitude: 24999549 };
- * const coords = new Coordinates('EPSG:4326', position.longitude, position.latitude, position.altitude); // Geographic system
- * const coordinates = coords.as('EPSG:4978'); // Geocentric system
- *
- * // or
- *
- * const position = { x: 20885167, y: 849862, z: 23385912 };
- * const coords = new Coordinates('EPSG:4978', position.x, position.y, position.z);  // Geocentric system
- * const coordinates = coords.as('EPSG:4326');  // Geographic system
- *
- * //or
- *
- * new Coordinates('EPSG:4326', longitude: 2.33, latitude: 48.24, altitude: 24999549).as('EPSG:4978'); // Geocentric system
- *
- * // or
- *
- * new Coordinates('EPSG:4978', x: 20885167, y: 849862, z: 23385912).as('EPSG:4326'); // Geographic system
- *
- * @param      {string} - [crs]{@link http://inspire.ec.europa.eu/theme/rs} : Geocentric (ex: 'EPSG:4326') or Geographic (ex: 'EPSG:4978').
- * @return     {Position} - position
- */
+        target.crs = crs;
 
-Coordinates.prototype.as = function as(crs, target) {
-    if (crs === undefined || CRS.toUnit(crs) === undefined) {
-        throw new Error(`Invalid crs paramater value '${crs}'`);
+        return target;
     }
-    return _convert(this, crs, target);
-};
 
-/**
- * Returns the normalized offset from top-left in extent of this Coordinates
- * e.g: extent.center().offsetInExtent(extent) would return (0.5, 0.5).
- * @param {Extent} extent
- * @param {Vector2} target optional Vector2 target. If not present a new one will be created
- * @return {Vector2} normalized offset in extent
- */
-Coordinates.prototype.offsetInExtent = function offsetInExtent(extent, target) {
-    if (this.crs != extent.crs) {
-        throw new Error('unsupported mix');
-    }
+    /**
+     * Returns the normalized offset from top-left in {@link Extent} of this
+     * Coordinates e.g: `extent.center().offsetInExtent(extent)` would return
+     * `(0.5, 0.5)`.
+     *
+     * @param {Extent} extent
+     * @param {Vector2} [target] - Optional Vector2 target. If not specified a
+     * new one will be created.
+     *
+     * @return {Vector2} Normalized offset in extent.
+     */
+    offsetInExtent(extent, target = new THREE.Vector2()) {
```

Comment:
May be move in `Extent Class`? what do you think?

Reply:
I moved it directly to `DEMUtils`, where it is the only place it used.

Reply:
ok